crates/next-dev/src/lib.rs

```diff
@@ -199,51 +211,70 @@ impl NextDevServerBuilder {
             log_level: self.log_level,
         };
         let entry_requests = Arc::new(self.entry_requests);
-        let console_ui = Arc::new(ConsoleUi::new(log_options));
-        let console_ui_to_dev_server = console_ui.clone();
         let server_addr = Arc::new(server.addr);
         let tasks = turbo_tasks.clone();
+        let issue_reporter_once = Arc::new(Lazy::new(|| {
+            self.issue_reporter
+                .unwrap_or_else(|| {
+                    // Initialize a ConsoleUi reporter if no custom reporter was provided
+                    Box::new(|log_options| ConsoleUi::new(log_options).cell().into())
+                })
+                .get_issue_reporter(log_options)
+        }));
+        let issue_reporter_arc = Arc::new(move || **issue_reporter_once);
```

Comment:
The Lazy is not needed as long `get_issue_reporter` is a turbo-tasks function. As it returns a Vc we should be able to assume that it's one.

Caching that with a Lazy will cause some parents to not be connected and this might cause weird effects.

You can remove the Lazy and rely on turbo-tasks functions.

You probably want to change:

``` diff
-ConsoleUi::new(log_options).cell()
+ConsoleUiVc::new(log_options)
```

So this is cached too and not creates cell over and over.